cve_bin_tool/cve_scanner.py

```diff
@@ -69,7 +70,23 @@ def get_cves(self, product_info: ProductInfo, triage_data: TriageData):
         """
         # Removing * from vendors that are guessed by the package list parser
         vendor = product_info.vendor.replace("*", "")
-        self.cursor.execute(query, [vendor, product_info.product, product_info.version])
+
+        # Need to manipulate version to ensure canonical form of version
+        if product_info.product == "openssl":
+            pv = re.search(r"\d[.\d]*[a-z]?", product_info.version)
+            parsed_version = parse_version(self.openssl_convert(pv.group(0)))
```

Comment:
Is there any risk of confusing the openssl parsing below (line 118 in your new file, line 105 in the original)

Reply:
@terriko It has taken me a while to see if there is a problem but I agree there may be an issue. If the version number in the version field of the database was an openssl number e.g. 1.1.1l then the change in new line 77 will result in the match being missed. I think the change should just call the openssl_convert for the comparison of versions (version_start_including etc) SQL statement and not for the first SQL statement although the manipulation of the version number still needs to be done so that 1.1.1l-1 becomes 1.1.1l.